lib/Bio/EnsEMBL/DataCheck/Checks/TranscriptVariation.pm

```diff
@@ -0,0 +1,72 @@
+=head1 LICENSE
+
+Copyright [2018-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::TranscriptVariation;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME        => 'TranscriptVariation',
+  DESCRIPTION => 'TranscriptVariation data is present and correct',
+  GROUPS      => ['variation_effect'],
+  DB_TYPES    => ['variation'],
+  TABLES      => ['transcript_variation ']
+};
+
+sub tests {
+  my ($self) = @_;
+
+  my $desc_1 = 'Peptide allele string not filled with digits';
+  my $sql_1 = qq/
+      SELECT COUNT(*)
+      FROM transcript_variation
+      WHERE pep_allele_string REGEXP '^[[:digit:]]+\$'
+  /;  
+  is_rows_zero($self->dba, $sql_1, $desc_1);
+  
+  my $desc_2 = 'Consequence type is not missing';
+  my $sql_2 = qq/
+      SELECT COUNT(*) 
+      FROM transcript_variation 
+      WHERE consequence_types = ''
+      OR consequence_types = 'NULL'
+      OR consequence_types IS NULL
+  /;
+  is_rows_zero($self->dba, $sql_2, $desc_2);
```

Comment:
PR #86 includes a new method which could be used here once that gets merged (currently called 'no_missing_values', but I suggested changing it to something else, maybe 'has_values').